streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -119,6 +122,24 @@ public void configure(Map<String, ?> configs) {
         streamThread = (StreamThread) o;
         streamThread.partitionAssignor(this);
 
+        String userEndPoint = (String) configs.get(StreamsConfig.USER_ENDPOINT_CONFIG);
+        if (userEndPoint != null && !userEndPoint.isEmpty()) {
+            final String[] hostPort = userEndPoint.split(":");
+            if (hostPort.length != 2) {
+                log.warn("Config '{}' isn't in the correct host:pair format. This instance will not be available for discovery"
```

Comment:
If user gives wrong configuration and later on cannot reach the stream instance, she need to look into the logs to figure out why the application is not reachable. This is not very user friendly IMHO.

But not sure what @enothereska @guozhangwang @miguno opinion is.